File: sdk/java/dagger-codegen-maven-plugin/src/main/java/io/dagger/codegen/introspection/TypeKind.java
package io.dagger.codegen.introspection;

public enum TypeKind {
  SCALAR("SCALAR"),
  OBJECT("OBJECT"),
  INTERFACE("INTERFACE"),
  UNION("UNION"),
  ENUM("ENUM"),
  INPUT_OBJECT("INPUT_OBJECT"),
  LIST("LIST"),
  NON_NULL("NON_NULL");

  private final String kind;

  TypeKind(String kind) {
    this.kind = kind;
  }
}

File: sdk/csharp/src/Dagger.SDK.CodeGen/Types/TypeKind.cs
using System.Text.Json.Serialization;

namespace Dagger.SDK.CodeGen.Types;

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum TypeKind
{
    SCALAR,
    OBJECT,
    INTERFACE,
    UNION,
    ENUM,
    INPUT_OBJECT,
    LIST,
    NON_NULL,
}
